File: src/main/java/gregicality/nuclear/api/capability/ICoolantHandler.java
package gregicality.nuclear.api.capability;

import net.minecraft.util.EnumFacing;
import net.minecraftforge.fluids.Fluid;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import gregicality.nuclear.api.capability.impl.LockableFluidTank;
import gregicality.nuclear.api.nuclear.fission.ICoolantStats;

public interface ICoolantHandler extends ILockableHandler<Fluid> {

    @Nullable
    ICoolantStats getCoolant();

    void setCoolant(@Nullable ICoolantStats prop);

    @NotNull
    LockableFluidTank getFluidTank();

    @NotNull
    EnumFacing getFrontFacing();
}

File: src/main/kotlin/io/github/symmetricdevs/supercritical/api/capability/ICoolantHandler.kt
package io.github.symmetricdevs.supercritical.api.capability

import io.github.symmetricdevs.supercritical.api.machine.multiblock.IFissionReactorHatch
import io.github.symmetricdevs.supercritical.api.machine.trait.LockableFluidTank
import net.minecraft.core.Direction
import net.minecraftforge.fluids.FluidStack

interface ICoolantHandler : ILockableHandler<FluidStack>, IFissionReactorHatch {

    val fluidTank: LockableFluidTank

    val coolantFrontFacing: Direction

    val outputHandler: ICoolantHandler?
}
